cfme-performance/utils/providers.py

```diff
@@ -8,19 +8,79 @@ def add_provider(provider):
     """Adds Provider via the Rest API."""
     logger.debug('Adding Provider: {}, Type: {}'.format(provider['name'], provider['type']))
 
+    data_dict = {
+        "action": "create",
+        "resources": [{
+            "name": provider['name'],
+            "type": provider['type'],
+            "hostname": provider['ip_address'],
+            "credentials": [{
+                "userid": provider['credentials']['username'],
+                "password": provider['credentials']['password']
+            }]
+        }]
+    }
+
+    if 'metrics credentials' in provider:
+        data_dict['resources'][0]['credentials'].append({
+            "userid": provider['metrics_credentials']['username'],
+            "password": provider['metrics_credentials']['password'],
+            "auth_type": "metrics"
+        })
+
+    # else if 'OTHERCREDENTIAL TYPES' in cfme_performance[provider]:
+    #     # TODO append the appropriate credentials
+
+    json_data = json.dumps(data_dict)
+    appliance = cfme_performance['appliance']['ip_address']
+    response = requests.post("https://" + appliance + "/api/providers",
+                             data=json_data,
+                             auth=(cfme_performance['appliance']['rest_api']['username'],
+                                   cfme_performance['appliance']['rest_api']['password']),
+                             verify=False,
+                             headers={"content-type": "application/json"},
+                             allow_redirects=False)
+
+    logger.debug('The response for adding Provider: {} is: {}'
+    .format(provider['name'], response.json()))
```

Comment:
Does .json produce json output into the log file?  We should just report the response code.